mopidy/ext.py

```diff
@@ -1,61 +1,71 @@
-import collections
+from __future__ import annotations
+
 import logging
 from collections.abc import Mapping
+from typing import TYPE_CHECKING, NamedTuple
 
 import pkg_resources
 
 from mopidy import config as config_lib
 from mopidy import exceptions
 from mopidy.internal import path
 
-logger = logging.getLogger(__name__)
+if TYPE_CHECKING:
+    from pathlib import Path
+    from typing import Any, Dict, Iterator, List, Optional, Type, Union
+
+    from mopidy.commands import Command
+    from mopidy.config import ConfigSchema
 
+    ConfigValue = Union[None, bool, float, int, str, int]
+    Config = Dict[str, Dict[str, ConfigValue]]
 
-_extension_data_fields = [
-    "extension",
-    "entry_point",
-    "config_schema",
-    "config_defaults",
-    "command",
-]
 
-ExtensionData = collections.namedtuple("ExtensionData", _extension_data_fields)
+logger = logging.getLogger(__name__)
+
+
+class ExtensionData(NamedTuple):
+    extension: "Extension"
+    entry_point: Any
+    config_schema: ConfigSchema
+    config_defaults: Any
+    command: Optional[Command]
 
 
 class Extension:
 
     """Base class for Mopidy extensions"""
 
-    dist_name = None
+    dist_name: str
     """The extension's distribution name, as registered on PyPI
 
     Example: ``Mopidy-Soundspot``
     """
 
-    ext_name = None
+    ext_name: str
     """The extension's short name, as used in setup.py and as config section
     name
 
     Example: ``soundspot``
     """
 
-    version = None
+    version: str
     """The extension's version
 
     Should match the :attr:`__version__` attribute on the extension's main
     Python module and the version registered on PyPI.
     """
 
-    def get_default_config(self):
+    def get_default_config(self) -> bytes:
```

Comment:
Shouldn't this be str? In all our bundled extensions we do:
```
    def get_default_config(self):
        conf_file = os.path.join(os.path.dirname(__file__), "ext.conf")
        return config.read(conf_file)
```
And our `config.read()` uses  `Path.read_text()` to return a `str` rather than `bytes`.

Reply:
Agree, will fix.